Applications/DataExplorer/DataView/RasterDataToMeshDialog.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "RasterDataToMeshDialog.h"
+#include "OGSError.h"
+#include "StrictDoubleValidator.h"
+
+#include <QSettings>
+#include <QFileDialog>
+
+RasterDataToMeshDialog::RasterDataToMeshDialog(std::string const& mesh_name,
+                                               QDialog* parent)
+    : QDialog(parent)
+{
+    setupUi(this);
+
+    this->meshNameEdit->setText(QString::fromStdString(mesh_name) + "_Raster");
+    auto* no_data_validator = new StrictDoubleValidator(this);
+    this->noDataValueEdit->setValidator (no_data_validator);
+}
+
+void RasterDataToMeshDialog::on_rasterSelectButton_pressed()
+{
+    QSettings settings;
+    QString filename = QFileDialog::getOpenFileName(
+        this, "Select raster file to open",
+        settings.value("lastOpenedRasterFileDirectory").toString(),
+        "ASCII raster files (*.asc);;All files (* *.*)");
+    this->rasterPathEdit->setText(filename);
+    QFileInfo fi(filename);
+    settings.setValue("lastOpenedRasterFileDirectory", fi.absolutePath());
+}
+
+void RasterDataToMeshDialog::accept()
+{
+    if (this->rasterPathEdit->text().isEmpty())
+    {
+        OGSError::box("Please specify path to raster file.");
+        return;
+    }
+    else if (this->arrayNameEdit->text().isEmpty())
```

Comment:
No else after a return needed. Will be a warning in clang-tidy.
```suggestion
    if (this->arrayNameEdit->text().isEmpty())
```

Reply:
This is correct the way it is. Otherwise the user drowns in error pop-ups.

Reply:
After OGSError::box is displayed, there is a return. Therefor the other if-conditions will not be checked and the user is presented with single error box.

Reply:
Exactly. This is the way it is supposed to work.